cfme/cloud/provider/__init__.py

```diff
@@ -309,41 +214,45 @@ def step(self, *args, **kwargs):
         sel.click(InfoBlock.element('Relationships', 'Images'))
 
 
-def get_all_providers(do_not_navigate=False):
+def get_all_providers():
     """Returns list of all providers"""
-    if not do_not_navigate:
-        navigate_to(CloudProvider, 'All')
-    providers = set([])
-    link_marker = "ems_cloud"
-    for page in paginator.pages():
-        for title in sel.elements("//div[@id='quadicon']/../../../tr/td/a[contains(@href,"
-                "'{}/show')]".format(link_marker)):
-            providers.add(sel.get_attribute(title, "title"))
-    return providers
-
-
-def discover(credential, cancel=False, d_type="Amazon"):
+    view = navigate_to(CloudProvider, 'All')
+    return [item.name for item in view.items.get_all(surf_pages=True)]
+
+
+def discover(credential, cancel=False, d_type="Amazon EC2"):
```

Comment:
Did we do this like this in the infra, we should pass in the provider type and then inside this method map that to a string, probably with a dict?